src/element.ts

```diff
@@ -37,7 +37,7 @@ export interface ElementStyle {
  */
 export abstract class Element {
   protected static ID: number = 1000;
-  protected context?: RenderContext;
+  private context?: RenderContext;
```

Comment:
you should order the fields. I would suggest
- public
- protected
- private

Otherwise it is a mess

Reply:
Agreed with this. We can fix in the next PR with comments. (Also maybe group it in some sane way.)